pyccel/parser/syntactic.py

```diff
@@ -451,13 +465,56 @@ def _visit_ImportNode(self, stmt):
                           bounding_box=stmt.absolute_bounding_box,
                           severity='error')
 
+        scope = stmt
+        while scope.parent != None and not isinstance(scope, (RedBaron, DefNode)):
```

Comment:
Line 458 checks that `stmt.parent` is an object of type `RedBaron` or `DefNode`, otherwise it creates an error report. Then line 463 creates an error report if the same object is of type `DefNode`. Hence `stmt.parent` must be of type `RedBaron` when we enter this cycle, or am I missing something here?

Reply:
You are correct. I missed this restriction